src/lib/precondition-resolvers/cooldown.ts

```diff
@@ -0,0 +1,42 @@
+import { container } from '@sapphire/pieces';
+import type { Command } from '../structures/Command';
+import { BucketScope, CommandPreConditions } from '../types/Enums';
+import { type PreconditionContainerArray } from '../utils/preconditions/PreconditionContainerArray';
+
+/**
+ * Appends the `Cooldown` precondition when {@link Command.Options.cooldownLimit} and
+ * {@link Command.Options.cooldownDelay} are both non-zero.
+ *
+ * @param command The command to parse cooldowns for.
+ * @param cooldownLimit The cooldown limit to use.
+ * @param cooldownDelay The cooldown delay to use.
+ * @param cooldownScope The cooldown scope to use.
+ * @param cooldownFilteredUsers The cooldown filtered users to use.
+ * @param preconditionContainerArray The precondition container array to append the precondition to.
+ */
+export function parseConstructorPreConditionsCooldown<P, O extends Command.Options>(
```

Comment:
Why not make it C extends Command<any, any>?

Reply:
Is that really better...? I'm unsure because you introduce any's.

Reply:
Why would you be afraid of `any`s lol, thats not bad. But it should work